index.ts

```diff
@@ -106,30 +101,87 @@ wss.on('connection', function connection(ws) {
+                    id: theUser.id,
+                    snum: sNum
+                };
+                knex("users")
+                    .insert(queryDetails)
+                    .catch((err) => {
+                        console.error(err);
+                        throw err;
+                    });
                 ws.userDetails = theUser;
                 const arrayClients = Array.from(wss.clients);
                 // @ts-ignore
                 const arrayUsers: Array<User> = Array.from(arrayClients, x => x.userDetails).filter(l => l != null);
-                users = arrayUsers.reduce((acc, elem) => {
-                    acc[elem.id] = elem;
-                    return acc;
-                }, {});
+                users = arrayUsers.reduce((acc, elem) => {acc[elem.id] = elem;return acc;}, {});
                 ws.send(JSON.stringify(["bestowId", theUser.id]));
                 sendToClients("newUser", users);
             }
             if (category === "loseUser") {
                 delete users[message.id];
                 delete users[ws.userId];
+                knex("users")
+                    .where({id: message.id})
+                    .del()
+                    .catch((err) => {
+                        console.error(err);
+                        throw err;
+                    });
+                knex("messages")
+                    .where({userid: message.id})
+                    .update({userid: 0})
+                    .catch((err) => {
+                        console.error(err);
+                        throw err;
+                    });
                 sendToClients("loseUser", users);
             }
             if (category === "newChannel") {
                 const channelId = ++highestChannelId;
-                channels[channelId] = new Channel (message.name, channelId, []);
+                const newChannel: Channel = {
+                    name: message.name,
+                    id: channelId,
+                    messages: []
+                };
+                channels[channelId] = newChannel;
+                knex("channels")
+                    .insert({name: newChannel.name, id: newChannel.id, messages: []})
+                    .catch((err) => {
+                        console.error(err);
+                        throw err;
+                    });
                 sendToClients("newChannel", channels);
             }
             if (category === "deleteChannel") {
                 delete channels[message];
+                knex("channels")
+                    .where({id: message})
+                    .del()
+                    .catch(err => {
+                        console.error(err);
+                        throw err;
+                    });
+                knex("messages")
+                    .where({channel: message})
+                    .del()
+                    .catch(err => {
+                        console.error(err);
+                        throw err;
```

Comment:
You have this piece of code in a lot of places, why not use await and use a big try catch block?

Reply:
i guess